pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -242,6 +242,31 @@ func (c *Controller) GetProxyServiceInstances(node *model.Proxy) ([]*model.Servi
 	return out, errs
 }
 
+func (c *Controller) GetProxyWorkloadLabels(proxy *model.Proxy) (model.LabelsCollection, error) {
+	out := make(model.LabelsCollection, 0)
+	var errs error
+	// It doesn't make sense for a single proxy to be found in more than one registry.
+	// TODO: if otherwise, warning or else what to do about it.
+	for _, r := range c.GetRegistries() {
+		labels, err := r.GetProxyWorkloadLabels(proxy)
+		if err != nil {
+			errs = multierror.Append(errs, err)
+		} else if len(labels) > 0 {
+			out = append(out, labels...)
+			break
+		}
+	}
+
+	if len(out) > 0 {
+		if errs != nil {
+			log.Warnf("GetProxyWorkloadLabels() found match but encountered an error: %v", errs)
```

Comment:
this is going to be super noisy. make it debug..

Reply:
Ok, I aligned with GetProxyServiceInstances which also warns in that case.
So is it normal that a Registry returns an error?
My understanding is that a registry returns nil when the proxy is unknown, and an error in case of a real error.

I'll change it to debug.

Reply:
Updated the PR and changed to debug.